packages/devtools_app/lib/src/provider/instance_viewer/eval.dart

```diff
@@ -9,33 +9,37 @@ import 'package:flutter_riverpod/flutter_riverpod.dart';
 
 import '../../eval_on_dart_library.dart';
 import '../../globals.dart';
+import '../../vm_service_wrapper.dart';
+
+/// Exposes the current VmServiceWrapper.
+/// By listening to this provider instead of directly accessing `serviceManager.service`,
+/// this ensures that providers reload properly when the devtool is connected
+/// to a different application.
+final serviceProvider = StreamProvider<VmServiceWrapper>((ref) async* {
```

Comment:
make this an instance member on serviceManager

Reply:
I am not following what this change involves.

Reply:
rather than making this a top level field, is there a reason not to make it a field on the serviceManager object?
As it is, this field is tightly coupled to what the serviceManager was it was first accessed so it is cleaner to have it be a member on the serviceManager itself anyway.